public/js/actions/tests/sources.js

```diff
@@ -0,0 +1,199 @@
+const fs = require ("fs");
+const path = require("path");
+const expect = require("expect.js");
+const { Task } = require("../../utils/task");
+
+// Mocks; these must come before anything else requires them, so they
+// are above all other requires.
+require("../../utils/networkRequest").networkRequest = function(url) {
+  return new Promise((resolve, reject) => {
+    // example.com is used at a dummy URL that points to our local
+    // `/public/js` folder.
+    url = url.replace("http://example.com/", "/../../");
+    resolve(JSON.parse(fs.readFileSync(__dirname + url, "utf8")));
+  });
+}
```

Comment:
Yeah, that's true. It became generic enough it probably could be. Usually test mock things themselves as needed. If we run into other things we want to mock we can think about it more, but I'll move it.